### PR TITLE
Remove metadata ID feature (previously deprecated in Orion 1.13.0)

### PR DESCRIPTION
**Pull request by fgalan**
Tasks:

- [x] Removal in the code, adjusting existing functional and unitarty tests
- [x] Add new .test to ensure the removal is correctly done, in particular:
   * Metadata ID works as any other metadata
   * Attemp to create an entity with a duplicated attribute name using NGSIv1 updateContext(), only differencing in metadata ID and get "duplication error"
   * Ensure that all the removed service routes return now a ~~405 Method Not Allowed error~~ "service not found" error
- [X] Provide a python script (and associated documentation) to check DB for existing metadata ID and warn about it

### CODE REVIEW

**Comment by fgalan** [comment]
This PR cannot be considered state. I contains the removal of metadata ID (deprecated since Orion 1.13.0) waiting for the right moment to remove it. It is upgraded from time to time with the master branch, so it is alive.

File: mkdocs.yml [hunk 1]
```diff
@@ -57,6 +57,7 @@ pages:
       - 'Upgrading from previous to 0.21.0': 'admin/upgrading_crossing_0-21-0.md'
       - 'Upgrading from previous to 1.3.0': 'admin/upgrading_crossing_1-3-0.md'
       - 'Upgrading from previous to 1.5.0': 'admin/upgrading_crossing_1-5-0.md'
+      - 'Upgrading from previous to 1.14.0': 'admin/upgrading_crossing_1-14-0.md'
```
Comment (fgalan) [inline mkdocs.yml]: Number (1.14.0) should be changed to the right one corresponding to the last version with metadata ID functionality, just before merging this PR.

File: doc/manuals/admin/upgrading_crossing_1-14-0.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+## Dealing with metadata ID
+
+In the case you have metadata ID in your database, you need to remove them. There are two ways of doing so:
+
+-   Using the Orion API with the old version of Orion you were using (recommended). You can use the
+    `DELETE /v1/contextEntities/{entityId}/attributes/{attrName}/{attrId}` or
+    `DELETE /v1/contextEntities/type/{entityType}/id/{entityId}/attributes/{attrName}/{attrId}` operations.
+
+-   Using the script check_metadata_id.py in autofix mode. In order to do so, edit the script and change the line `autofix = False`
```
Comment (fgalan) [inline doc/manuals/admin/upgrading_crossing_1-14-0.md]: Seems old, given that code has two posibilities for autofix:

```
autofix = None
#autofix = 'as_new_attrs'
#autofix = 'as_metadata'
```

Have a look and fix.

File: mkdocs.yml (hunk 1 above)
Comment (fgalan) [reply mkdocs.yml]: Fixed in 344a6bb

File: doc/manuals/admin/upgrading_crossing_1-14-0.md (hunk 2 above)
Comment (fgalan) [reply doc/manuals/admin/upgrading_crossing_1-14-0.md]: Fixed in 344a6bb

File: doc/manuals/admin/upgrading_crossing_2-2-0.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        WARNING: some problem was found during the process.
+        ERROR: document ... change attempt failed
+
+-   Install new contextBroker version. (Sometimes the commands fails due to yum cache. In that case, run "yum clean all" and try again)
+
+        yum install contextBroker
+
+-   Start contextBroker
+
+Note that the rpm command demands superuser privileges, so you have to run it as root or using the sudo command.
```
Comment (fisuda) [inline doc/manuals/admin/upgrading_crossing_2-2-0.md]: the `yum` command demands superuser privileges

File: doc/manuals/admin/upgrading_crossing_2-2-0.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+In the case you have metadata ID in your database, you need to remove them. There are two ways of doing so:
+
+-   Using the Orion API with the old version of Orion you were using (recommended). You can use the
+    `DELETE /v1/contextEntities/{entityId}/attributes/{attrName}/{attrId}` or
+    `DELETE /v1/contextEntities/type/{entityType}/id/{entityId}/attributes/{attrName}/{attrId}` operations.
+
+-   Using the script `check_metadata_id.py` in autofix mode. In order to do so, edit the script and change the line `autofix = None`
+    to either `autofix = 'as_new_attrs'` or `autofix = 'as_metadata'`.
+    - In `as_new_attrs` mode each attribute with ID is transformed to a new attribute in the form `<attrName>:<id>` (e.g. if you
+      have an attribute `temperature` with metadata ID `id1` then it will be transformed to `temperature:id1`. In the rare case the
```
Comment (fisuda) [inline doc/manuals/admin/upgrading_crossing_2-2-0.md]: - Add a close bracket
transformed to `temperature:id1`**)**.

File: doc/manuals/admin/upgrading_crossing_2-2-0.md (hunk 3 above)
Comment (fgalan) [reply doc/manuals/admin/upgrading_crossing_2-2-0.md]: Fixed in e97e48f1c4ef7b7f3b2bf995eed923df1cbc4ee9. Thanks!

File: doc/manuals/admin/upgrading_crossing_2-2-0.md (hunk 4 above)
Comment (fgalan) [reply doc/manuals/admin/upgrading_crossing_2-2-0.md]: Fixed in e97e48f1c4ef7b7f3b2bf995eed923df1cbc4ee9. Thanks!

**Comment by fgalan** [comment]
@fisuda this PR touches existing .md files and create also some new ones. I would be merged sooner or later (I think it will be soon :) so probably is a good idea to create a "parallel" PR with the Japanese synchronization, so I can merge both when the moment comes, please.

Thanks!

**Comment by fisuda** [comment]
Noted with thanks. I will send a new PR for Japanese translation.

File: doc/manuals/admin/install.md [hunk 5]
```diff
@@ -70,13 +70,16 @@ Upgrade procedure depends on whether the *upgrade path* (i.e. from the installed
 
 ### Migrating the data stored in DB
 
-You only need to pay attention to this if your upgrade path crosses 0.14.1, 0.19.0, 0.21.0 or 1.3.0. Otherwise, you can skip this section. You can also skip this section if your DB are not valuable (e.g. debug/testing environments) and you can flush your DB before upgrading.
+You only need to pay attention to this if your upgrade path crosses 0.14.1, 0.19.0, 0.21.0, 1.3.0, 1.5.0 and 1.14.0.
```
Comment (fgalan) [inline doc/manuals/admin/install.md]: 1.14.0 -> 2.2.0

@fisuda please sync in PR https://github.com/telefonicaid/fiware-orion/pull/3436

File: doc/manuals/admin/upgrading_crossing_2-2-0.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+-   Stop contextBroker
+-   Remove previous contextBroker version
+
+        yum remove contextBroker
+
+-   [Take a backup of your DBs](database_admin.md#backup) (this is just a safety measure in case any problem occurs,
+    e.g some script gets interrupted before finished and your database data ends in an incoherent state)
+-   Download the following script:
+    -   [check_metadata_id.py](https://github.com/telefonicaid/fiware-orion/blob/2.2.0/scripts/managedb/check_metadata_id.py)
```
Comment (fgalan) [inline doc/manuals/admin/upgrading_crossing_2-2-0.md]: https://github.com/telefonicaid/fiware-orion/blob/2.2.0/scripts/managedb/check_metadata_id.py -> https://github.com/telefonicaid/fiware-orion/blob/2.2.0/scripts/managedb/upgrade-2.2.0/check_metadata_id.py 

@fisuda please sync in PR #3436

**Comment by fisuda** [comment]
Noted with thanks.

File: doc/manuals/admin/install.md (hunk 5 above)
Comment (fgalan) [reply doc/manuals/admin/install.md]: Fixed in c21dd86

File: doc/manuals/admin/upgrading_crossing_2-2-0.md (hunk 6 above)
Comment (fgalan) [reply doc/manuals/admin/upgrading_crossing_2-2-0.md]: Fixed in c21dd86

**Comment by fgalan** [comment]
Overpassed by _identical_ PR https://github.com/telefonicaid/fiware-orion/pull/3445

**Pull request closed without merging**